Dockerfile

```diff
@@ -32,35 +32,28 @@ ARG USER="dagu"
 ARG USER_UID=1000
 ARG USER_GID=$USER_UID
 
-# Create user and set permissions
-RUN apk update; \
-    apk add --no-cache sudo tzdata; \
-    addgroup -g ${USER_GID} ${USER}; \
-    adduser ${USER} -h /home/${USER} -u ${USER_UID} -G ${USER} -D -s /bin/ash; \
-    echo ${USER} ALL=\(root\) NOPASSWD:ALL > /etc/sudoers.d/${USER}; \
-    chmod 0440 /etc/sudoers.d/${USER}; \
-    mkdir -p .config/dagu/dags; \
-    chown -R ${USER}:${USER} /home/${USER};
-
 COPY --from=go-builder /app/bin/dagu /usr/local/bin/
+COPY ./entrypoint.sh /entrypoint.sh
+
+# Create user and set permissions
+RUN apk update && \
+  apk add --no-cache su-exec shadow tzdata && \
+  addgroup -g ${USER_GID} ${USER} && \
+  adduser ${USER} -h /config -u ${USER_UID} -G ${USER} -D -s /bin/ash && \
+  chown -R ${USER}:${USER} /config && \
+  chmod +x /entrypoint.sh;
```

Comment:
Instead of `sudo` we rely on `su-exec` and allowing the user to run scripts in `/etc/custom-init.d` 

Ideally `sudo` would not be added to docker containers, since it is easy to get root if you need it anyway (through docker exec)